File: app/src/main/java/quangnam/com/sample/base/MvpFragment.java
package quangnam.com.sample.base;

import android.content.Context;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.annotation.StringRes;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.widget.Toast;

import javax.inject.Inject;

import dagger.android.AndroidInjector;
import dagger.android.DispatchingAndroidInjector;
import dagger.android.support.AndroidSupportInjection;
import dagger.android.support.HasSupportFragmentInjector;
import quangnam.com.base.exception.BaseException;
import quangnam.com.sample.di.ActivityContext;
import quangnam.com.sample.di.ChildFragmentManager;
import quangnam.com.sample.di.PerFragment;

/**
 * Created by quangnam on 11/23/17.
 * Project Sample
 */

public class MvpFragment extends BaseFragment implements IBaseView,
        HasSupportFragmentInjector      // *** NOTE:Use only if has child fragment use dagger
{

    /**
     * NOTE: Use if has child fragment use dagger
     */
    @Inject
    DispatchingAndroidInjector<Fragment> mChildFragmentInjector;

    @ActivityContext
    @Inject
    protected Context mActivityContext;

    @PerFragment
    @ChildFragmentManager
    @Inject
    protected FragmentManager mChildFragmentManager;

    /**
     * NOTE: Use if has child fragment use dagger
     */
    @Override
    public AndroidInjector<Fragment> supportFragmentInjector() {
        return mChildFragmentInjector;
    }

    @Override
    public void onActivityCreated(@Nullable Bundle savedInstanceState) {
        super.onActivityCreated(savedInstanceState);
        AndroidSupportInjection.inject(this);
    }

    @Override
    public void showLoading() {
        //TODO implement
    }

    @Override
    public void hideLoading() {
        // TODO
    }

    @Override
    public void onError(@NonNull BaseException exception) {
        Toast.makeText(mActivityContext, exception.getLocalizedMessage(), Toast.LENGTH_SHORT).show();
    }

    @Override
    public void onError(@StringRes int resID) {
        // TODO
    }

    @Override
    public void onError(String message) {
        // TODO
    }

    @Override
    public void onErrorCode(int errorCode) {
        // TODO
    }

    @Override
    public void hideKeyboard() {
        // TODO
    }
}

File: app/src/main/java/quangnam/com/sample/base/MvpFragment.kt
package quangnam.com.sample.base

import android.content.Context
import android.os.Bundle
import android.support.annotation.StringRes
import android.support.v4.app.Fragment
import android.support.v4.app.FragmentManager
import android.widget.Toast
import dagger.android.AndroidInjector
import dagger.android.DispatchingAndroidInjector
import dagger.android.support.AndroidSupportInjection
import dagger.android.support.HasSupportFragmentInjector
import quangnam.com.base.exception.BaseException
import quangnam.com.sample.di.ActivityContext
import quangnam.com.sample.di.ChildFragmentManager
import quangnam.com.sample.di.PerFragment
import javax.inject.Inject

/**
 * Created by quangnam on 11/23/17.
 * Project Sample
 */

open class MvpFragment : BaseFragment(), IBaseView, HasSupportFragmentInjector      // *** NOTE:Use only if has child fragment use dagger
{

    /**
     * NOTE: Use if has child fragment use dagger
     */
    @Inject
    lateinit var mChildFragmentInjector: DispatchingAndroidInjector<Fragment>

    @field:ActivityContext
    @Inject
    lateinit var mActivityContext: Context

    @PerFragment
    @field:ChildFragmentManager
    @Inject
    lateinit var mChildFragmentManager: FragmentManager

    /**
     * NOTE: Use if has child fragment use dagger
     */
    override fun supportFragmentInjector(): AndroidInjector<Fragment>? {
        return mChildFragmentInjector
    }

    override fun onActivityCreated(savedInstanceState: Bundle?) {
        super.onActivityCreated(savedInstanceState)
        AndroidSupportInjection.inject(this)
    }

    override fun showLoading() {
        //TODO implement
    }

    override fun hideLoading() {
        // TODO
    }

    override fun onError(exception: BaseException) {
        Toast.makeText(mActivityContext, exception.localizedMessage, Toast.LENGTH_SHORT).show()
    }

    override fun onError(@StringRes resID: Int) {
        // TODO
    }

    override fun onError(message: String) {
        // TODO
    }

    override fun onErrorCode(errorCode: Int) {
        // TODO
    }

    override fun hideKeyboard() {
        // TODO
    }
}
